thorlcr/activities/selfjoin/thselfjoinslave.cpp

```diff
@@ -236,6 +238,10 @@ class SelfJoinSlaveActivity : public CSlaveActivity, public CThorDataLink
         CriticalBlock b(joinHelperCrit);
         rowcount_t p = joinhelper?joinhelper->getLhsProgress():0;
         mb.append(p);
+
+        CRuntimeStatisticCollection mergedStats(spillStats);
+        mergeStats(mergedStats, sorter);
```

Comment:
Is there a potential race condition here between serializeStats() being called and reset(), or is it guaranteed that serializeStats() cannot be called when reset() is being called?